OsmAnd/src/net/osmand/plus/views/mapwidgets/widgetstates/ZoomLevelWidgetState.java

```diff
@@ -0,0 +1,81 @@
+package net.osmand.plus.views.mapwidgets.widgetstates;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.settings.backend.ApplicationMode;
+import net.osmand.plus.settings.backend.preferences.OsmandPreference;
+import net.osmand.plus.views.mapwidgets.WidgetType;
+import net.osmand.util.Algorithms;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringRes;
+
+public class ZoomLevelWidgetState extends WidgetState {
+
+	private final OsmandPreference<ZoomLevelType> typePreference;
+
+	public ZoomLevelWidgetState(@NonNull OsmandApplication app, @Nullable String customId) {
+		super(app);
+		typePreference = registerTypePreference(customId);
+	}
+
+	@NonNull
+	@Override
+	public String getTitle() {
+		return app.getString(getZoomLevelType().titleId);
+	}
+
+	@Override
+	public int getSettingsIconId(boolean nightMode) {
+		return WidgetType.DEV_ZOOM_LEVEL.getIconId(nightMode);
+	}
+
+	@Override
+	public void changeToNextState() {
+		typePreference.set(getZoomLevelType().next());
+	}
+
+	@Override
+	public void copyPrefs(@NonNull ApplicationMode appMode, @Nullable String customId) {
+		registerTypePreference(customId).setModeValue(appMode, getZoomLevelType());
```

Comment:
We can use the already registered typePreference

Reply:
Don't we need preference with `customId`?

Reply:
Yes, it's needed here